packages/driver/src/cypress/log.ts

```diff
@@ -349,33 +349,64 @@ export class Log {
       next: null,
     })
 
-    const snapshot = this.cy.createSnapshot(name, this.get('$el'))
+    let snapshot
+    const postProcessSnapshot = (shouldRebindSnapshotFn = true) => {
+      const snapshots = this.get('snapshots') || []
+
+      // don't add snapshot if we couldn't create one, which can happen
+      // if the snapshotting process errors
+      // https://github.com/cypress-io/cypress/issues/15816
+      if (snapshot) {
+      // insert at index 'at' or whatever is the next position
+        snapshots[options.at || snapshots.length] = snapshot
+      }
 
-    const snapshots = this.get('snapshots') || []
+      this.set('snapshots', snapshots)
 
-    // don't add snapshot if we couldn't create one, which can happen
-    // if the snapshotting process errors
-    // https://github.com/cypress-io/cypress/issues/15816
-    if (snapshot) {
-      // insert at index 'at' or whatever is the next position
-      snapshots[options.at || snapshots.length] = snapshot
+      if (options.next && shouldRebindSnapshotFn) {
+        const fn = this.snapshot
+
+        this.snapshot = function () {
+        // restore the fn
+          this.snapshot = fn
+
+          // call orig fn with next as name
+          return fn.call(this, options.next)
+        }
+      }
+
+      return this
     }
 
-    this.set('snapshots', snapshots)
+    if (this.config('experimentalSessionAndOrigin') && !Cypress.isCrossOriginSpecBridge) {
```

Comment:
Would it be possible to somehow tie this to the command group the log is part of? Like add the originPolicy to the command group and use it for any events generated within that group? @emilyrohrbough, is something like that possible (or better)?